### PR TITLE
[wasm] Bump emscripten to 3.1.9

### PR DESCRIPTION
**Pull request by radekdoulik**

### CODE REVIEW

File: src/native/libs/CMakeLists.txt [hunk 1]
```diff
@@ -144,6 +144,7 @@ if (CLR_CMAKE_TARGET_UNIX OR CLR_CMAKE_TARGET_BROWSER)
         add_compile_options(-Wno-typedef-redefinition)
         add_compile_options(-Wno-c11-extensions)
         add_compile_options(-Wno-thread-safety-analysis)
+        add_compile_options(-Wno-strict-prototypes)
```
Comment (lambdageek) [inline src/native/libs/CMakeLists.txt:147]: is this about zlib?

**Comment by lewing** [comment]
@radekdoulik the packages are in https://github.com/dotnet/runtime/pull/69127 so please prioritize this

File: src/native/libs/CMakeLists.txt (hunk 1 above)
Comment (radekdoulik) [reply src/native/libs/CMakeLists.txt:147]: > If it's about zlib, I ran into something like this with trying to bump us to 1.2.12. Ended up adding a function `set_zlib_source_files_properties` to `src/native/external/zlib.cmake` to turn off strict prototypes just for those source files: [3c3d19b](https://github.com/dotnet/runtime/commit/3c3d19b06532f565c617dae18d7fb23e9eafdab5)

Part of the warnings come indeed from zlib IIRC, there are many more in other parts as well.

I have opened https://github.com/dotnet/runtime/issues/69142 to not forget to fix them.

**Comment by radekdoulik** [comment]
helix failures, array length:
```
[22:21:59] info: Discovered:  System.IO.MemoryMappedFiles.Tests.dll (found 73 of 223 test cases)
[22:21:59] info: Using random seed for test cases: 1666391570
[22:21:59] info: Using random seed for collections: 1666391570
[22:21:59] info: Starting:    System.IO.MemoryMappedFiles.Tests.dll
[22:22:00] fail: RangeError: Invalid typed array length: 9223372036854776000
[22:22:00] fail:     at new Uint8Array (<anonymous>)
[22:22:00] fail:     at Object.resizeFileStorage (./dotnet.js:12:29362)
[22:22:00] fail:     at Object.setattr (./dotnet.js:12:30202)
[22:22:00] fail:     at Object.truncate (./dotnet.js:12:47406)
[22:22:00] fail:     at Object.ftruncate (./dotnet.js:12:47603)
[22:22:00] fail:     at ___syscall_ftruncate64 (./dotnet.js:12:85703)
[22:22:00] fail:     at ftruncate (<anonymous>:wasm-function[8913]:0x1ccdd)9
[22:22:00] fail:     at SystemNative_FTruncate (<anonymous>:wasm-function[8759]:0x1c24f)7
[22:22:00] fail:     at wasm_invoke_iil (<anonymous>:wasm-function[8111]:0x1a570)b
[22:22:00] fail:     at interp_exec_method (<anonymous>:wasm-function[171]:0xe8c)8
[22:22:00] fail:     at interp_runtime_invoke (<anonymous>:wasm-function[170]:0xcf0)b
[22:22:00] fail:     at mono_jit_runtime_invoke (<anonymous>:wasm-function[7836]:0x19715)c
[22:22:00] fail:     at do_runtime_invoke (<anonymous>:wasm-function[1975]:0x84a2)5
[22:22:00] fail:     at mono_runtime_invoke_checked (<anonymous>:wasm-function[1973]:0x8499)b
[22:22:00] fail:     at ves_icall_InternalInvoke_raw (<anonymous>:wasm-function[1319]:0x6767)9
[22:22:00] fail:     at do_icall (<anonymous>:wasm-function[264]:0x1d33)5
[22:22:00] fail:     at do_icall_wrapper (<anonymous>:wasm-function[234]:0x1c7b)8
[22:22:00] fail:     at interp_exec_method (<anonymous>:wasm-function[171]:0xe0c)5
[22:22:00] fail:     at interp_runtime_invoke (<anonymous>:wasm-function[170]:0xcf0)b
[22:22:00] fail:     at mono_jit_runtime_invoke (<anonymous>:wasm-function[7836]:0x19715)c
[22:22:00] fail:     at do_runtime_invoke (<anonymous>:wasm-function[1975]:0x84a2)5
[22:22:00] fail:     at mono_runtime_try_invoke (<anonymous>:wasm-function[1979]:0x8508)5
[22:22:00] fail:     at mono_runtime_invoke (<anonymous>:wasm-function[2006]:0x8715)b
[22:22:00] fail:     at mono_wasm_invoke_method_ref (<anonymous>:wasm-function[115]:0xa07d)
[22:22:00] fail:     at Module._mono_wasm_invoke_method_ref (./dotnet.js:12:123542)
[22:22:00] fail:     at Object.o.<computed> (./dotnet.js:3:4986)
[22:22:00] fail:     at _call_method_with_converted_args (./dotnet.js:3:55076)
[22:22:00] fail:     at call_method_ref (./dotnet.js:3:54218)
[22:22:00] fail:     at ./dotnet.js:3:56062
[22:22:00] fail:     at mono_call_assembly_entry_point (./dotnet.js:3:56194)
[22:22:00] fail:     at Object.mono_run_main (./dotnet.js:3:66153)
[22:22:00] fail:     at Object.init (test-main.js:229:43)
[22:22:00] fail:     at Object.onDotnetReady (test-main.js:173:17)
[22:22:00] fail:     at finalize_startup (./dotnet.js:3:72326)
[22:22:00] fail:     at mono_wasm_after_runtime_initialized (./dotnet.js:3:68522)
[22:22:00] fail:     at callRuntimeCallbacks (./dotnet.js:12:17280)
[22:22:00] fail:     at postRun (./dotnet.js:12:13252)
[22:22:00] fail:     at doRun (./dotnet.js:12:140605)
[22:22:00] fail:     at run (./dotnet.js:12:140762)
[22:22:00] fail:     at runCaller (./dotnet.js:12:140242)
[22:22:00] fail:     at removeRunDependency (./dotnet.js:12:14044)
[22:22:00] fail:     at receiveInstance (./dotnet.js:12:15837)
[22:22:00] fail:     at receiveInstantiationResult (./dotnet.js:12:15958)
[22:22:00] info: Process v8 exited with 1
```
network failure (maybe related to previous array length?):
```
[22:23:17] info: Discovered:  System.Net.Http.Functional.Tests.dll (found 387 of 1620 test cases)
[22:23:17] info: Using random seed for test cases: 967384189
[22:23:17] info: Using random seed for collections: 967384189
[22:23:17] info: Starting:    System.Net.Http.Functional.Tests.dll
[22:23:21] fail: (node:6277) Warning: Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by disabling certificate verification.
[22:23:21] fail: (Use `node --trace-warnings ...` to show where the warning was created)
[22:23:21] fail: [FAIL] System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_HttpClientHandlerTest_Headers_Http11.SendAsync_ContentLengthAndTransferEncodingHeaders_IgnoreContentLength
[22:23:21] info: System.Net.Http.HttpRequestException : FetchError: request to http://127.0.0.1:35635/ failed, reason: Parse Error: Content-Length can't be present with Transfer-Encoding
[22:23:21] info: ---- System.Runtime.InteropServices.JavaScript.JSException : FetchError: request to http://127.0.0.1:35635/ failed, reason: Parse Error: Content-Length can't be present with Transfer-Encoding
[22:23:21] info:    at System.Net.Http.BrowserHttpHandler.<SendAsync>g__Impl|56_0(HttpRequestMessage request, CancellationToken cancellationToken)
[22:23:21] info:    at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)
[22:23:21] info:    at System.Net.Http.Functional.Tests.HttpClientHandlerTest_Headers.<>c__DisplayClass20_0.<<SendAsync_ContentLengthAndTransferEncodingHeaders_IgnoreContentLength>b__1>d.MoveNext()
[22:23:21] info: --- End of stack trace from previous location ---
[22:23:21] info:    at System.Net.Test.Common.LoopbackServer.AcceptConnectionAsync(Func`2 funcAsync)
[22:23:21] info:    at System.Net.Http.Functional.Tests.HttpClientHandlerTest_Headers.<SendAsync_ContentLengthAndTransferEncodingHeaders_IgnoreContentLength>b__20_0(LoopbackServer server, Uri uri)
[22:23:21] info:    at System.Net.Test.Common.LoopbackServer.CreateServerAsync(Func`2 funcAsync, Options options)
[22:23:21] info:    at System.Net.Http.Functional.Tests.HttpClientHandlerTest_Headers.SendAsync_ContentLengthAndTransferEncodingHeaders_IgnoreContentLength()
[22:23:21] info: --- End of stack trace from previous location ---
[22:23:21] info: ----- Inner Stack Trace -----
[22:23:21] info:    at System.Net.Http.BrowserHttpHandler.<SendAsync>g__Impl|56_0(HttpRequestMessage request, CancellationToken cancellationToken)
[22:23:23] info: Finished:    System.Net.Http.Functional.Tests.dll
[22:23:23] info: 
[22:23:23] info: === TEST EXECUTION SUMMARY ===
[22:23:23] info: Total: 595, Errors: 0, Failed: 1, Skipped: 72, Time: 5.818907s
[22:23:23] info: 
[22:23:23] info: Received expected 295 of /datadisks/disk1/work/B72A09D5/w/A5FA0997/uploads/xharness-output/testResults.xml
[22:23:23] info: Finished writing 295 bytes of RESULTXML
[22:23:23] info: Xml file was written to the provided writer.
[22:23:23] info: Tests run: 595 Passed: 522 Inconclusive: 0 Failed: 1 Ignored: 0 Skipped: 72
[22:23:23] info: Process node exited with 1
```
timeout:
```
[00:08:02] info: Incoming arguments: --run WasmTestRunner.dll System.AppContext.Tests.dll -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing
[00:08:02] info: Application arguments: --run WasmTestRunner.dll System.AppContext.Tests.dll -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing
[00:08:02] info: console.debug: mono_wasm_runtime_ready fe00e07a-5519-4dfe-b35a-f867dbaf2e28
[00:08:02] info: console.debug: Could not find symbols file dotnet.js.symbols. Ignoring.
[00:08:02] info: console.info: Initializing.....
[00:08:02] info: Discovering: System.AppContext.Tests.dll (method display = ClassAndMethod, method display options = None)
[00:08:02] info: Discovered:  System.AppContext.Tests.dll (found 15 of 15 test cases)
[00:08:02] info: Using random seed for test cases: 1302376283
[00:08:02] info: Using random seed for collections: 1302376283
[00:08:02] info: Starting:    System.AppContext.Tests.dll
[00:08:02] info: Finished:    System.AppContext.Tests.dll
[00:08:02] info: 
[00:08:02] info: === TEST EXECUTION SUMMARY ===
[00:08:02] info: Total: 15, Errors: 0, Failed: 0, Skipped: 0, Time: 0.0921059s
[00:08:02] info: 
[00:08:02] info: Received expected 3166 of C:\helix\work\workitem\uploads\xharness-output\testResults.xml
[00:08:02] info: Finished writing 3166 bytes of RESULTXML
[00:08:02] info: Xml file was written to the provided writer.
[00:08:02] info: Tests run: 15 Passed: 15 Inconclusive: 0 Failed: 0 Ignored: 0 Skipped: 0
[00:23:01] info: Process 5012 didn't exit within 00:15:00 and will be killed

[00:23:01] info: Killing process tree of 5012...

[00:23:01] info: Pids to kill: 5012

[00:23:01] info: Running lldb diagnostics for pid 5012

[00:23:01] info: Printing backtrace for pid=5012

[00:23:01] info: lldb was not found, skipping diagnosis..

[00:23:01] info: 5012 Execution timed out after 900 seconds and the process was killed.

[00:23:01] info: Process v8.cmd exited with -1

[00:23:01] fail: Application has finished with exit code -1 but 0 was expected
XHarness exit code: 71 (GENERAL_FAILURE)
```
another timeout:
```
Using existing build found at C:\helix\work\workitem\e\szdkcnim.3vr, with build log at C:\helix\work\workitem\uploads\xharness-output\logs\szdkcnim.3vr\non_async_main_args_Release_False.binlog
    Wasm.Build.Tests.MainWithArgsTests.NonAsyncMainWithArgs(buildArgs: BuildArgs { ProjectName = placeholder, Config = Release, AOT = False, ProjectFileContents = placeholder, ExtraBuildArgs =  }, args: ["abc", "foobar"], host: V8, id: "szdkcnim.3vr") [FAIL]
      Process timed out. Last 20 lines of output:
      
      [szdkcnim.3vr]   Restored C:\helix\work\workitem\e\szdkcnim.3vr\non_async_main_args_Release_False.csproj (in 99 ms).
      
      [szdkcnim.3vr] C:\helix\work\correlation\dotnet-workload\sdk\7.0.100-preview.5.22260.5\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.RuntimeIdentifierInference.targets(216,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy [C:\helix\work\workitem\e\szdkcnim.3vr\non_async_main_args_Release_False.csproj]
      
      [szdkcnim.3vr]   non_async_main_args_Release_False -> C:\helix\work\workitem\e\szdkcnim.3vr\bin\Release\net7.0\browser-wasm\non_async_main_args_Release_False.dll
      
      [szdkcnim.3vr]   Optimizing assemblies for size. This process might take a while.
      
      [szdkcnim.3vr]   ** MicrosoftNetCoreAppRuntimePackDir : C:\helix\work\correlation\dotnet-workload\packs\Microsoft.NETCore.App.Runtime.Mono.browser-wasm\7.0.0-ci
      
      [szdkcnim.3vr]   non_async_main_args_Release_False -> C:\helix\work\workitem\e\szdkcnim.3vr\bin\Release\net7.0\browser-wasm\publish\
      
      [szdkcnim.3vr]   non_async_main_args_Release_False -> C:\helix\work\workitem\e\szdkcnim.3vr\bin\Release\net7.0\browser-wasm\non_async_main_args_Release_False.dll
      
      [szdkcnim.3vr]   ** MicrosoftNetCoreAppRuntimePackDir : C:\helix\work\correlation\dotnet-workload\packs\Microsoft.NETCore.App.Runtime.Mono.browser-wasm\7.0.0-ci
      
      [szdkcnim.3vr]   non_async_main_args_Release_False -> C:\helix\work\workitem\e\szdkcnim.3vr\bin\Release\net7.0\browser-wasm\publish\
```

**Comment by radekdoulik** [comment]
OSX failure is https://github.com/dotnet/runtime/issues/69190

**Pull request closed without merging**